[FIX] account, l10n_cl : move_type invisible condition eval

loleloup opened the pull request:
steps:
with studio, try adding a field to the invoice tree view -> traceback

This is due to the 'default_move_type' context being a string but trying to be evaluated as a boolean.

opw-3703559
opw-3702620
opw-3702360
opw-3698111
opw-3698108
opw-3702084

robodoo commented:
[Pull request status dashboard](https://mergebot.odoo.com/odoo/odoo/pull/151293).

william-andre commented on addons/account/views/account_move_views.xml:
```diff
@@ -435,7 +435,7 @@
                            optional="show"/>
                     <field name="state" widget="badge" decoration-success="state == 'posted'" decoration-info="state == 'draft'" optional="show"/>
-                    <field name="move_type" invisible="context.get('default_move_type', True)"/>
+                    <field name="move_type" invisible="context.get('default_move_type', False) == False"/>
```
At the very worst
```suggestion
                    <field name="move_type" invisible="bool(context.get('default_move_type', True))"/>
```

But that really seems more like a framework issue, a truthy value should be castable to a bool directly in the framework.

clbr-odoo reviewed (changes requested):
Thank you for your PR,
I think this is no longer needed in 17.0+ thanks to this https://github.com/odoo/odoo/pull/104741 :thinking:

vin-odoo reviewed (changes requested):
As said on discord to angv, please ensure that the issue is not happening before 16.3. Also a similar fix is needed for [l10n_cl](https://github.com/odoo/odoo/blob/saas-16.3/addons/l10n_cl/views/account_move_view.xml#L66) then.

As WAN said, please first check with the framework if this isn't an issue on their side.

I also believe you are changing the behavior of the invisible, see the table with another solution:

|'default_move_type' in context|Value of 'default_move_type'|context.get('default_move_type', True)|context.get('default_move_type', False) == False|context.get('default_move_type', True) != False|
|-|-|-|-|-|
|Yes|False|False|True|False|
|Yes|'entry'|'entry'|False|True|
|No|N/A|True|True|True|

Edit: I took too long to type my table :joy:

loleloup commented:
@william-andre @clbr-odoo @vin-odoo 

Thanks for all the comments, didn't expect so much stuff to happen while I was off :)

I updated the PR with the bool suggestion (how did I not think of that dammit) and added the l10n_cl view in the fix.

For a bit more context : the issue appears since [this PR](https://github.com/odoo/enterprise/pull/54552) that adds the context when adding a field to a view.  I asked about it in framework python but got a reply that got deleted and chm seems a bit off, I will insist on this issue later on.  In the meantime, I believe this is a pretty small fix that won't have side-effects and there are a lot of tickets because of this, so it should be ok to merge for 16.3 and 16.4.

Let me know what you think and thanks for your time

william-andre commented:
I will merge this on Friday if we have no other fix since.
Fixing it in the view will require an update anyway... But fixing it in the JS, where it should won't.

kebeclibre commented:
https://github.com/odoo/odoo/pull/153022/files @william-andre @loleloup

loleloup commented:
fixed by https://github.com/odoo/enterprise/pull/56372
